### PR TITLE
Delete Item

### LINKED ISSUE #12: 12. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future
## Summary
Users might make a mistake when entering an item, or may decide they won’t be buying a certain item again in the future. Allowing them to delete list items will help them keep their lists tidy.
 
## Acceptance criteria
- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Notes:
* Activating the “Delete” button should show a confirmation dialog. We want to give users a chance to confirm “destructive” actions.
* It might be tempting to create your own modal dialog component to handle confirmation, but modals are fraught with accessibility concerns, so consider using a simple [JavaScript confirm dialog](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm).

### PR DESCRIPTION
**Pull request by MarcosPerez16**
## Description

Added delete button and implemented a function. Created a confirm window for the user upon deleting item. Filled out deleteItem function in firebase

The code adds a delete button and `deleteItem` function as well as a prompt asking if the user would like to delete the item. We learned more about how our firebase db structure and layers work. 

## Related Issue

closes #12  

## Acceptance Criteria

- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Type of Changes

`feature`

### After

screenshot for issue 12
Screenshot for issue 12 number 2

## Testing Steps / QA Criteria

1. Go to a list, click on the delete button.
2. Confirm whether or not you would like to delete
3. Item should be deleted on screen, and in firebase

### CODE REVIEW

**Review by arandel1 — COMMENTED** [review]
Looks good and works properly. Nice job!

File: src/components/ListItem.jsx [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+import { updateItem, deleteItem } from '../api';
 
 export function ListItem({ item }) {
-	const { name, dateLastPurchased, dateCreated, id } = item;
+	const { name, dateLastPurchased, id } = item;
 	const [checked, setChecked] = useState(false);
 
+	const handleDelete = async () => {
+		try {
+			if (window.confirm(`Are you sure you want to delete ${name}?`)) {
```
Comment (sar-mko) [inline src/components/ListItem.jsx:11]: I like the check here :) and the use of async/await

**Review by sar-mko — APPROVED** [review]
Looks good! I got the prompt, it deleted on the screen, and updates in firebase.

**Review by nathanejbrown — APPROVED** [review]

**Review by mentalcaries — APPROVED** [review]
Nice clean implementation. Good job!

**Pull request merged**